### PR TITLE
[OSDEV-1005] Propagate facility deletions to OpenSearch data source

### PR DESCRIPTION
**Pull request by vladsha-dev**
[[OSDEV-1005](https://opensupplyhub.atlassian.net/browse/OSDEV-1005)] Propagate facility deletions to OpenSearch data source

- Implement the propagation of production location deletions from the PostgreSQL database to the OpenSearch cluster. After this fix, the locations that were deleted will be excluded from the response of the `v1/production-location` GET API endpoint.
- Enable deployment of the Logstash and OpenSearch infra to AWS environments.
- Disconnect location deletion propagation to the OpenSearch cluster while the Django tests are running, as it is outside the scope of Django unit testing.

[OSDEV-1005]: https://opensupplyhub.atlassian.net/browse/OSDEV-1005?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: deployment/terraform/firewall.tf [hunk 1]
```diff
@@ -425,3 +435,13 @@ resource "aws_security_group_rule" "opensearch_ingress" {
   security_group_id        = aws_security_group.opensearch.id
   source_security_group_id = aws_security_group.app_logstash.id
 }
+
+resource "aws_security_group_rule" "app_opensearch_ingress" {
```
Comment (VadimKovalenkoSNF) [inline deployment/terraform/firewall.tf:439]: What problem this ingress rule solved? Afair, there were no problems with incoming traffic to the OpenSearch previously in https://github.com/opensupplyhub/open-supply-hub/pull/220.

File: src/django/api/signals.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+from django.db.models.signals import post_delete
+from django.dispatch import receiver
+from opensearchpy.exceptions import ConnectionError
+
+from api.models.facility.facility import Facility
+from api.services.opensearch import OpenSearchServiceConnection
+from oar.rollbar import report_error_to_rollbar
+
+
+# Initialize logger.
```
Comment (VadimKovalenkoSNF) [inline src/django/api/signals.py:13]: This comment can be removed

File: src/django/api/signals.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+log = logging.getLogger(__name__)
+
+
+@receiver(post_delete, sender=Facility)
+def location_post_delete_handler_for_opensearch(sender, **kwargs):
+    opensearch = OpenSearchServiceConnection()
+    location_instance = kwargs.get('instance')
+    try:
+        response = opensearch.client.delete(
+            index='production-locations',
```
Comment (VadimKovalenkoSNF) [inline src/django/api/signals.py:23]: In the future I'd like to have these index values listed in a separate file as constants.

**Review by VadimKovalenkoSNF — CHANGES_REQUESTED** [review]
Overall looks good for me, but left few notes before approval.

File: src/django/api/signals.py (hunk 2 above)
Comment (vladsha-dev) [reply src/django/api/signals.py:13]: Done

File: deployment/terraform/firewall.tf (hunk 1 above)
Comment (vladsha-dev) [reply deployment/terraform/firewall.tf:439]: It is necessary to enable a network connection from Django to OpenSearch. Previously, ingress traffic from Django to OpenSearch was blocked.

**Review by VadimKovalenkoSNF — APPROVED** [review]
LGTM

**Pull request merged**